internal/provider/data_source_auth_methods.go

```diff
@@ -0,0 +1,170 @@
+					Type:        schema.TypeString,
+					Description: "Output only. The ID of the Auth Method.",
+					Computed:    true,
+				},
+				"is_primary": {
+					Type:        schema.TypeBool,
+					Description: "Output only. Whether this auth method is the primary auth method for it's scope.\nTo change this value update the primary_auth_method_id field on the scope.",
+					Computed:    true,
+				},
+				"name": {
+					Type:        schema.TypeString,
+					Description: "Optional name for identification purposes.",
+					Computed:    true,
+				},
+				"scope": {
+					Type:     schema.TypeList,
+					Computed: true,
+					Elem: &schema.Resource{
+						Schema: map[string]*schema.Schema{
+							"description": {
+								Type:        schema.TypeString,
+								Description: "Output only. The description of the Scope, if any.",
+								Computed:    true,
+							},
+							"id": {
+								Type:        schema.TypeString,
+								Description: "Output only. The ID of the Scope.",
+								Computed:    true,
+							},
+							"name": {
+								Type:        schema.TypeString,
+								Description: "Output only. The name of the Scope, if any.",
+								Computed:    true,
+							},
+							"parent_scope_id": {
+								Type:        schema.TypeString,
+								Description: "Output only. The ID of the parent Scope, if any. This field will be empty if this is the \"global\" scope.",
+								Computed:    true,
+							},
+							"type": {
+								Type:        schema.TypeString,
+								Description: "Output only. The type of the Scope.",
+								Computed:    true,
+							},
+						},
+					},
+				},
+				"scope_id": {
+					Type:        schema.TypeString,
+					Description: "The ID of the Scope of which this Auth Method is a part.",
+					Computed:    true,
+				},
+				"type": {
+					Type:        schema.TypeString,
+					Description: "The Auth Method type.",
+					Computed:    true,
+				},
+				"updated_time": {
+					Type:        schema.TypeString,
+					Description: "Output only. The time this resource was last updated.",
+					Computed:    true,
+				},
+				"version": {
+					Type:        schema.TypeInt,
+					Description: "Version is used in mutation requests, after the initial creation, to ensure this resource has not changed.\nThe mutation will fail if the version does not match the latest known good version.",
+					Computed:    true,
+				},
+			},
+		},
+	},
+	"recursive": {
+		Type:     schema.TypeBool,
+		Optional: true,
+	},
+	"scope_id": {
+		Type:     schema.TypeString,
+		Optional: true,
+	},
+}
+
+func dataSourceAuthMethods() *schema.Resource {
+	return &schema.Resource{
+		Description: "Lists all Auth Methods.",
+		Schema:      dataSourceAuthMethodsSchema,
+		ReadContext: dataSourceAuthMethodsRead,
+	}
+}
+
+func dataSourceAuthMethodsRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client := meta.(*metaData).client
+
+	req, err := client.NewRequest(ctx, "GET", "auth-methods", nil)
```

Comment:
Is the reason for crafting this without calling the [API package for authmethods](https://github.com/hashicorp/boundary/blob/main/api/authmethods/authmethods.gen.go#L86) to make generating the codebase easier?